### PR TITLE
feat: adds external account authorized user credentials

### PR DESCRIPTION
**Pull request by lsirac**
Adds support for external account authorized user credentials. 

See go/gcloud-3pi-headful.

### CODE REVIEW

File: oauth2_http/java/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentials.java [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+        .setQuotaProjectId(quotaProjectId)
+        .build();
+  }
+
+  private void readObject(ObjectInputStream input) throws IOException, ClassNotFoundException {
+    input.defaultReadObject();
+    transportFactory = newInstance(transportFactoryClassName);
+  }
+
+  private boolean canRefresh() {
```
Comment (TimurSadykov) [inline oauth2_http/java/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentials.java:336]: MArk as VisibleForTesting and cover?

File: oauth2_http/java/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentials.java [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+        && refreshToken.trim().length() > 0
+        && tokenUrl != null
+        && tokenUrl.trim().length() > 0
+        && clientId != null
+        && clientId.trim().length() > 0
+        && clientSecret != null
+        && clientSecret.trim().length() > 0;
+  }
+
+  private HttpRequest buildRefreshRequest() throws IOException {
```
Comment (TimurSadykov) [inline oauth2_http/java/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentials.java:347]: Mark as VisibleForTesting and cover?

File: oauth2_http/java/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentials.java [hunk 3]
```diff
@@ -0,0 +1,513 @@
...
+        tokenUrl,
+        tokenInfoUrl,
+        revokeUrl,
+        audience,
+        transportFactoryClassName,
+        quotaProjectId);
+  }
+
+  @Override
+  public String toString() {
```
Comment (TimurSadykov) [inline oauth2_http/java/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentials.java:259]: test?

File: oauth2_http/javatests/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentialsTest.java [hunk 4]
```diff
@@ -0,0 +1,821 @@
...
+      ExternalAccountAuthorizedUserCredentials secondCredentials =
+          ExternalAccountAuthorizedUserCredentials.fromJson(json, transportFactory);
+      assertNotEquals(secondCredentials, credentials);
+      assertNotEquals(credentials, secondCredentials);
+      assertNotEquals(credentials.hashCode(), secondCredentials.hashCode());
+    }
+  }
+
+  @Test
+  public void equals_differentCredentials() throws IOException {
```
Comment (TimurSadykov) [inline oauth2_http/javatests/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentialsTest.java:759]: add equals tests for each property like in ServiceAccountCredential tests? Sorry for the churn, but should be relatively fast to copy-past them.

**Review by TimurSadykov — CHANGES_REQUESTED** [review]
Overall LGTM, only test coverage comments

File: oauth2_http/java/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentials.java (hunk 1 above)
Comment (lsirac) [reply oauth2_http/java/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentials.java:336]: This is sufficiently tested through the public API surface.

File: oauth2_http/java/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentials.java (hunk 2 above)
Comment (lsirac) [reply oauth2_http/java/com/google/auth/oauth2/ExternalAccountAuthorizedUserCredentials.java:347]: Same as above - this is covered.

**Review by TimurSadykov — APPROVED** [review]
LGTM, but we might fix codecoverage later

**Pull request merged**